client/lib/site-spec/utils.ts

```diff
@@ -0,0 +1,71 @@
+import config from '@automattic/calypso-config';
+
+// Raw config structure from the server
+interface SiteSpecRawConfig {
+	agent_url?: string;
+	agent_id?: string;
+	build_site_url?: string;
+	script_url: string;
+	css_url: string;
+}
+
+// Config structure for the client
+interface SiteSpecConfig {
+	agentUrl?: string;
+	agentId?: string;
+	buildSiteUrl?: string;
+}
+
+// Config key for URL functions
+type UrlKey = 'script_url' | 'css_url';
+
+// Resource type mapping
+export type ResourceType = 'script' | 'css';
+
+/**
+ * Checks if the SiteSpec feature is enabled in the current environment.
+ * @returns True if SiteSpec is enabled, false otherwise
+ */
+export function isSiteSpecEnabled(): boolean {
+	return config.isEnabled( 'site-spec' );
+}
+
+/**
+ * Retrieves the cache-busted URL for a specific SiteSpec resource.
+ * @param urlKey - The configuration key for the resource type ({@link UrlKey}, default: 'script_url')
+ * @returns The complete URL with cache-busting parameters, or null if not configured
+ */
+export function getSiteSpecUrl( urlKey: UrlKey = 'script_url' ): string | null {
+	const siteSpecConfig = config( 'site_spec' ) as SiteSpecRawConfig | undefined;
+	const url = siteSpecConfig?.[ urlKey ];
+
+	return url || null;
```

Comment:
Maybe worth doing this:

```suggestion
	return url?.trim() ?? null;
```